### PR TITLE
Add Args.count()

### PR DESCRIPTION
**Pull request by CohenArthur**
- parser: WIP: Parse empty types -> Read message
- empty_types: Add empty type parsing and instantiation
- empty_types: Instantiate empty types on the fly
- stdlib: Add Args.count() function
- stdlib: Make Args type an empty type

### CODE REVIEW

File: tests/ft/type_checking/valid/empty_types.jk [hunk 1]
```diff
@@ -0,0 +1,17 @@
+type Empty;
+e = Empty;
+
+type Ok;
+type NotOk;
+
+o = Ok;
```
Comment (Skallwar) [inline tests/ft/type_checking/valid/empty_types.jk:7]: I'm not sure if this is a good design. Is `o` the type `Ok` or an instance of `Ok` (I know it's the latter but that a bit confusing).

Why not just go with the usual `Ok()` rather than creating a special edge case that complexify the language.

What is the benefit of this outside of removing the need of `()`

Comment (CohenArthur) [reply tests/ft/type_checking/valid/empty_types.jk:7]: Oops... that's my bad for having the wrong destination branch, but this discussion is more suited on #348. I'll answer there
